### PR TITLE
Add data.interactive example handler

### PR DESCRIPTION
**Pull request by wbamberg**
This PR lints `data.interactive example?` according to the specification: https://github.com/mdn/stumptown-content/blob/master/project-docs/javascript-linter-spec.md#datainteractive_example, as requested in https://github.com/mdn/stumptown-content/issues/408.

I wanted to use a couple of functions that were in other handlers: `isNewlineOnlyTextNode` and `isAdmonition`, so moved them into utils.

### CODE REVIEW

**Comment by wbamberg** [comment]
I've updated this to return a position node from the handler: https://github.com/mdn/stumptown-content/pull/416/commits/0515f38f68ebb6d7f67105eb337c5c191b223f4e. I moved `isAdmonition` back out of utils since we're not using it anywhere else now.

I didn't update tests to test position nodes yet, because I expect to want to use some utilities that are in the [`data.constructor` PR](https://github.com/mdn/stumptown-content/pull/421).

File: scripts/scraper-ng/test/ingredient-data.interactive_example.test.js
Comment (ddbeck) [inline scripts/scraper-ng/test/ingredient-data.interactive_example.test.js:43]: Why do these tests assert the unknown heading message? Since they exercise another rule, I'm not sure how they increase our confidence in this ingredient's handler. How about asserting that it _doesn't_ have a `data.interactive_example?` message? Something like…

```js
    expect(file).not.hasMessageWithId("data.interactive_example?");
```

**Review by ddbeck — CHANGES_REQUESTED** [review]
Hey Will, this is a nice start. Interactive examples more complicated than I imagined! I made a bunch of line comments, though there's not actually any major problems to report here. Thank you!

File: scripts/scraper-ng/rules/html-require-recipe-ingredients/ingredient-handlers/index.js
Comment (wbamberg) [reply scripts/scraper-ng/rules/html-require-recipe-ingredients/ingredient-handlers/index.js:27]: they are in a mad order that makes no sense

File: scripts/scraper-ng/rules/html-require-recipe-ingredients/ingredient-handlers/data-interactive-example.js
Comment (wbamberg) [reply scripts/scraper-ng/rules/html-require-recipe-ingredients/ingredient-handlers/data-interactive-example.js:9]: Given the other changes, nothing ended up calling this function so I just removed it.

File: scripts/scraper-ng/rules/html-require-recipe-ingredients/ingredient-handlers/data-interactive-example.js
Comment (wbamberg) [reply scripts/scraper-ng/rules/html-require-recipe-ingredients/ingredient-handlers/data-interactive-example.js:65]: I've made this change.

File: scripts/scraper-ng/rules/html-require-recipe-ingredients/ingredient-handlers/data-interactive-example.js
Comment (wbamberg) [reply scripts/scraper-ng/rules/html-require-recipe-ingredients/ingredient-handlers/data-interactive-example.js:83]: I wasn't quite sure what you meant here, but I ended up `visit`ing them here, so I can log errors for every extra call at the pint it's made. But I also wanted to hang onto the node where I found the call.

File: scripts/scraper-ng/rules/html-require-recipe-ingredients/ingredient-handlers/data-interactive-example.js
Comment (wbamberg) [reply scripts/scraper-ng/rules/html-require-recipe-ingredients/ingredient-handlers/data-interactive-example.js:91]: Since we're logging errors as we find extra calls, there's no `> 1` case any more.

File: scripts/scraper-ng/test/ingredient-data.interactive_example.test.js
Comment (wbamberg) [reply scripts/scraper-ng/test/ingredient-data.interactive_example.test.js]: This is tricky though where there are actual interactive example errors. What I want to say, I suppose, is: there are _only these particular_ `data.interactive_example?` errors, and I have tried to cobble together a regex to that effect.

**Comment by wbamberg** [comment]
Thanks for the review comments @ddbeck ! I've tried to address them here.

**Review by ddbeck — APPROVED** [review]
Hey Will, your revisions have been great—I like this a lot! I'm marking this as approved, but if you want to sort `ingredientHandlers` before merging, I don't need to review that. Or you can merge it as is. Either way, thank you. 🎉

**Comment by wbamberg** [comment]
Oh, I forgot about the reordering, that's done now. Thanks as usual for the excellent for the review, Daniel!

**Pull request merged**
